### PR TITLE
Prevent HTML from being entered into Edit Data Cells.

### PR DESCRIPTION
**Pull request by smartguest**
Addresses https://github.com/microsoft/azuredatastudio/issues/20881

Prevents Submission of a cell that contains HTML code, also has commented out option to clean the code immediately for submitting, whichever is preferred.

Whatever is entered in the text editor remains part of the data of the grid (until the table is manually refreshed with server values in safe cell format) and switching tabs will result in the HTML executing upon rerender with no way to change this behavior from ADS's side aside from whatever is in the applyValue function. 

A proposed idea that I have is to immediately refresh/reload the table (akin to clicking run) upon submit so that a Display Edit Data Cell (which is safe upon rerender) will replace the cell containing live HTML. But this will require testing to see if it works or not (including when switching tabs) and if there are any side effects.

This is a small workaround for now.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
Is this linked to the correct issue? It's unclear to me how HTML content relates to that issue

**Comment by smartguest** [comment]
@Charles-Gagnon Fixed it

File: src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts [hunk 1]
```diff
@@ -1001,14 +1004,20 @@ export class EditDataGridPanel extends GridParentComponent {
...
-				let colIndex = self.getColumnIndex(this._args.column.name);
 				let dataLength: number = self.dataSet.dataRows.getLength();
 
 				// If this is not the "new row" at the very bottom
 				if (activeRow !== dataLength) {
-					currentRow[colIndex] = state;
-					this._textEditor.applyValue(item, state);
+					// Prevent applying if the cell contains HTML as it will execute automatically due to slickgrid's design:
+					// the slickgrid text editor's value affects what's in the grid's data, so that means entering HTML will result in execution upon immediate rerender.
+					let containsHTML = /<([A-Za-z][A-Za-z0-9]*)\b[^>]*>(.*?)<\/\1>/.test(state);
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:1013]: Does this prevent HTML that's already in the database from causing issues?

Comment (smartguest) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:1013]: Generally, HTML already in the database will be in the safe cell format that does not execute upon tab switch rerender, so if it is altered in any way by the user, the code will handle it correctly.

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:1013]: Generally? When would it not be?

@alanrenmsft @aasimkhan30 You're familiar with this slickgrid stuff, is there any ideas you have here for how to handle this better? Regex to detect HTML and blocking it seems like a not-great user experience.

Comment (smartguest) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:1013]: Poor choice of words, it is always in the safe cell format when retrieved directly from the server, it is only immediately after making an edit that the local cell is a not safe format.

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:1013]: Gotcha, so as long as the user doesn't actually try to change the value then it's fine? Do you know why the rendering done for that is different than the rendering done when we submit a change?

**Review by kburtram — CHANGES_REQUESTED** [review]
I don't think preventing inputting HTML is a good fix for this issue.  Can we update SlickGrid to not render the HTML that is typed into the input box?

**Comment by smartguest** [comment]
Closing PR, making changes to slickgrid instead

**Pull request closed without merging**